components/script/dom/dommatrix.rs

```diff
@@ -0,0 +1,343 @@
+        fn M43(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().M43()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-m43
+        fn SetM43(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m43(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-m44
+        fn M44(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().M44()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-m44
+        fn SetM44(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m44(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-a
+        fn A(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().A()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-a
+        fn SetA(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m11(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-b
+        fn B(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().B()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-b
+        fn SetB(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m12(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-c
+        fn C(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().C()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-c
+        fn SetC(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m21(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-d
+        fn D(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().D()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-d
+        fn SetD(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m22(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-e
+        fn E(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().E()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-e
+        fn SetE(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m41(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-f
+        fn F(&self) -> f64 {
+            self.upcast::<DOMMatrixReadOnly>().F()
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-f
+        fn SetF(&self, value: f64) {
+            self.upcast::<DOMMatrixReadOnly>().set_m42(value);
+        }
+
+        // https://drafts.fxtf.org/geometry-1/#dom-dommatrix-multiplyself
+        fn MultiplySelf(&self, other:&DOMMatrixInit) -> Root<DOMMatrix> {
```

Comment:
Ok. But if I'm going to add `matrix` and `is_2d` accessors to `DOMMatrix` anyway, then I can  just move the contents of the `*Self` methods to `DOMMatrixReadOnly` and get rid of the `*_self` methods from `DOMMatrix`. It will be overall less code if that works.